casbin/enforcer.h

```diff
@@ -1,26 +1,106 @@
-/*
-* Copyright 2020 The casbin Authors. All Rights Reserved.
-*
-* Licensed under the Apache License, Version 2.0 (the "License");
-* you may not use this file except in compliance with the License.
-* You may obtain a copy of the License at
-*
-*    http://www.apache.org/licenses/LICENSE-2.0
-*
-* Unless required by applicable law or agreed to in writing, software
-* distributed under the License is distributed on an "AS IS" BASIS,
-* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-* See the License for the specific language governing permissions and
-* limitations under the License.
-*/
-
 #pragma once
 
-#include <string>
+#ifdef CASBIN_EXPORTS
+#define ENFORCER_API __declspec(dllexport)
+#else
+#define ENFORCER_API __declspec(dllimport)
+#endif
+
+#include "./model/model.h"
+#include "./model/function.h"
+#include "./persist/file-adapter/file-adapter.h"
+#include "./persist/adapter_filtered.h"
+#include "./persist/adapter.h"
+#include "./util/util.h"
+#include "./effect/effect.h"
+#include <initializer_list>
+#include <unordered_map>
+#include <variant>
+#include"third_party/Cparse/shunting-yard.h"
+
+
+#define REGISTER_START                                  \
+public:                                                 \
+    unordered_map<string,packToken> GetMap() {          \
+        unordered_map<string,packToken> tm;
+
+#define REGISTER_MEMBER(_MEMBER_)                       \
+        tm[#_MEMBER_] = _MEMBER_ ;
+
+#define REGISTER_END                                    \
+        return tm;                                      \
+    };

```

Comment:
Naming should improve so other people can know it's only for ABAC, Also are these macros necessary?